01_easy_algorithms/06_delete_extra_elements/06_delete_extra_elements.rb

```diff
@@ -0,0 +1,16 @@
+def delete_occurrences (arr, num)
+  element_count = Hash.new(0)
+  delete_arr = []
+  new_arr = []
+  arr.each do |n|
+    element_count[n] += 1
+    
+    if element_count[n] > num
+      delete_arr << n
+    else
+      new_arr << n
+    end
```

Comment:
```suggestion
    if element_count[n] <= num
      new_arr << n
    end
```

You don't need to have an else condition on your if block. Since you're not using the delete_arr for anything and you're only adding to it so that you have something to do in the if condition, you can remove it altogether and add to new_arr if a condition you give it is true.

You can condense that further into a one liner without making it to confusing for the reader:
`new_arr << n if element_count[n] <= num`